File: AI/BFS 8 puzzle/puzzle/src/BFS/Node.java
package BFS;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Rakshith on 2/2/2018.
 */
public class Node {

    private int data;
    private List<Node> neighbours;
    private boolean visited;


    public Node(int data) {
        this.data = data;
        neighbours = new ArrayList<>();
    }

    public int getData() {
        return data;
    }

    public void setData(int data) {
        this.data = data;
    }

    public List<Node> getNeighbours() {
        return neighbours;
    }

    public void setNeighbours(List<Node> neighbours) {
        this.neighbours = neighbours;
    }

    public boolean isVisited() {
        return visited;
    }

    public void setVisited(boolean visited) {
        this.visited = visited;
    }

    public void addNeighbour(Node neighbour){
        neighbours.add(neighbour);
    }

    @Override
    public String toString() {
        return String.valueOf(data);
    }
}

File: AI/BFS 8 puzzle/bfs python/Basicbfs/Node.py
class Node:
    def __init__(self, data):
        self.data = data
        self.neighbours = []
        self.visited = False

    def add_neighbour(self, neighbour):
        self.neighbours.append(neighbour)

    def preview_neighbours(self):
        print(self.neighbours)

    def get_neighbours(self):
        return self.neighbours

    def set_visited(self, truth):
        self.visited = truth

    def get_data(self):
        return self.data

    def is_visited(self):
        return self.visited

    def __eq__(self, other):
        return isinstance(other, Node) and self.data == other.data

    def __str__(self):
        return str(self.data)
